DiamondLightSource/python-artemis
# 129 opentelemetry tracing

Fixes #129
Adds an OpenTelemetry tracer to time different steps of the plan, which we can easily export to prometheus or any other database/ui for monitoring

### To test:
1. For testing, run the Jaeger all-in-one container: https://www.jaegertracing.io/docs/1.6/getting-started/
2. Run artemis in dev/S03/fake zocalo mode and run through the plan. you may need to wrap do_fgs in a try block and throw away the odin/filewriter waittimeouterror
3. Confirm tracing results show up in Jaeger UI, maybe add some more `with tracer.start_span("span_name"):` in various places
